mmv1/products/alloydb/User.yaml

```diff
@@ -0,0 +1,112 @@
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/alloydb_user.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'alloydb_user_builtin'
+    primary_resource_id: 'user1'
+    vars:
+      alloydb_cluster_name: 'alloydb-cluster'
+      alloydb_cluster_pass: 'cluster_secret'
+      alloydb_cluster_resource_id: 'default'
+      alloydb_instance_name: 'alloydb-instance'
+      alloydb_instance_resource_id: 'default'
+      alloydb_user_name: 'user1'
+      alloydb_user_pass: 'user_secret'
+      network_name: 'alloydb-network'
+    test_vars_overrides:
+      alloydb_cluster_name: '"tf-test-user-cluster-" + acctest.RandString(t, 10)'
+      alloydb_cluster_pass: '"cluster_secret1"'
+      alloydb_cluster_resource_id: '"test-cluster1"'
+      alloydb_instance_name: '"tf-test-instance1"'
+      alloydb_instance_resource_id: '"test-instance1"'
+      alloydb_user_name: '"user_1"'
+      alloydb_user_pass: '"user_1_pass"'
+      network_name: 'acctest.BootstrapSharedTestNetwork(t, "alloydb-user-builtin")'
+    ignore_read_extra:
+      - 'password'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'alloydb_user_iam'
+    primary_resource_id: 'user2'
+    vars:
+      alloydb_cluster_name: 'alloydb-cluster'
+      alloydb_cluster_resource_id: 'default'
+      alloydb_instance_name: 'alloydb-instance'
+      alloydb_instance_resource_id: 'default'
+      alloydb_cluster_pass: 'cluster_secret'
+      alloydb_user_name: 'user2@foo.com'
+      network_name: 'alloydb-network'
+    test_vars_overrides:
+      alloydb_cluster_name: '"tf-test-user-cluster-" + acctest.RandString(t, 10)'
+      alloydb_cluster_pass: '"cluster_secret2"'
+      alloydb_cluster_resource_id: '"test-cluster2"'
+      alloydb_instance_name: '"tf-test-instance2"'
+      alloydb_instance_resource_id: '"test-instance2"'
+      alloydb_user_name: '"user2@foo.com"'
+      network_name: 'acctest.BootstrapSharedTestNetwork(t, "alloydb-user-iam")'
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'cluster'
+    description: |
+      Identifies the alloydb cluster. Must be in the format
+      'projects/{project}/locations/{location}/clusters/{cluster_id}'
+    required: true
+    immutable: true
+    resource: 'Cluster'
+    imports: 'name'
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'userId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The database role name of the user.
+  - !ruby/object:Api::Type::Enum
+    name: 'userType'
+    required: true
+    immutable: true
+    description: |
+      The type of this user.
+    values:
+      - :ALLOYDB_BUILT_IN
+      - :ALLOYDB_IAM_USER
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'password'
+    ignore_read: true
+    description: |
+      Password for this database user.
+  - !ruby/object:Api::Type::Array
+    name: 'databaseRoles'
```

Comment:
Can we have a test coverage on multiple entries of this field. Just want to make sure if the API returns the values in the same order as they're sent.

Reply:
Valid point. I am waiting for an AlloyDB control plane deployment that would return the database roles in sorted order. That is exactly why I didn't create a test case, yet. I will definitely add one the moment the deployment is finished.

Reply:
I think we shouldn't block the TF feature because TF in essence should reflect the API behaviour (even if it's buggy).

We should aim to submit the test specifically for this purpose when the control plane changes are rolled out to prod to fix this issue.
I have created a bug for the same and assigned to @pattukerman . b/301552557

Reply:
I understand your point but the same issue will come when someone tries to hit a normal POST API as well.
It's not something that's inherently a TF problem but an API issue and we should at least give the users the capability to use TF to provision and manage IAM users. Once the fix is applied, TF and API both will work as expected.

Reply:
The change on the AlloyDB Control Plane side to return the database roles in sorted order is almost fully deployed to production. The deployment will finish over the weekend.

Reply:
The change is fully deployed to prod.